quickwit/quickwit-proto/src/lib.rs

```diff
@@ -297,11 +296,60 @@ pub fn set_parent_span_from_request_metadata(request_metadata: &tonic::metadata:
     Span::current().set_parent(parent_cx);
 }
 
+/// Index identifiers that uniquely identify not only the index, but also
+/// its incarnation allowing to distinguish between deleted and recreated indexes.
+/// It is represented as a stiring in index_id:incarnation_id format.
+#[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq, Eq, Ord, PartialOrd, Hash)]
+pub struct IndexUid(String);
+
+impl IndexUid {
+    /// Creates a new index uid form index_id and incarnation_id
+    pub fn new(index_id: impl ToString) -> Self {
+        Self::from_parts(index_id,  Ulid::new().to_string())
```

Comment:
`Ulid` are 26-char long, but we actually don't need that many random chars because of the index ID prefix. Should we only use a few random characters to increase readiblity?

Reply:
Yeah, we can probably safely cut it in half considering frequency of index creation.